### ISSUE TITLE
[SDXL] 客户使用stable_diffusion_xl时出现的问题

### ISSUE THREAD

**Issue by LinF00** [mindspore-lab/mindone#135]
一：现在sdxl情况是推理一张图片要15分钟，模型加载要将近2分钟，推理时间太久了，需要优化
二：然后finetune和lora训练都初始化出错，无法训练

**Comment by zhanghuiyao** [comment]
问题一：(1) 第一个step中会存在编译时间，一遍会比较长，可以从第二个step开始统计下时间；(2) 推荐使用streamlit 方式运行；
问题二：(1) 当前仅支持lora finetune，vanilla finetune还存在显存问题；(2) lora finetune的报错可以贴一下；

**Comment by zhanghuiyao** [comment]
程序运行开始的时候都会进行编译，用streamlit方式运行，第一张图片比较慢，后面的会变快

**Comment by zhanghuiyao** [comment]
在线推理的性能会持续优化，后续计划也会有离线推理版本；

**Comment by LinF00** [comment]
训练时报错日志
[train.log](https://github.com/mindspore-lab/mindone/files/12592659/train.log)

flash attention is available.
config:{'model': {'target': 'gm.models.diffusion.DiffusionEngine', 'params': {'scale_factor': 0.13025, 'disable_first_stage_amp': False, 'denoiser_config': {'target': 'gm.modules.diffusionmodules.denoiser.DiscreteDenoiser', 'params': {'num_idx': 1000, 'weighting_config': {'target': 'gm.modules.diffusionmodules.denoiser_weighting.EpsWeighting'}, 'scaling_config': {'target': 'gm.modules.diffusionmodules.denoiser_scaling.EpsScaling'}, 'discretization_config': {'target': 'gm.modules.diffusionmodules.discretizer.LegacyDDPMDiscretization'}}}, 'network_config': {'target': 'gm.modules.diffusionmodules.openaimodel.UNetModel_lora', 'params': {'lora_dim': 4, 'lora_alpha': 4, 'lora_dropout': 0.0, 'lora_merge_weights': False, 'only_save_lora': True, 'adm_in_channels': 2816, 'num_classes': 'sequential', 'in_channels': 4, 'out_channels': 4, 'model_channels': 320, 'attention_resolutions': [4, 2], 'num_res_blocks': 2, 'channel_mult': [1, 2, 4], 'num_head_channels': 64, 'use_spatial_transformer': True, 'use_linear_in_transformer': True, 'transformer_depth': [1, 2, 10], 'context_dim': 2048, 'spatial_transformer_attn_type': 'flash-attention', 'legacy': False, 'use_recompute': True}}, 'conditioner_config': {'target': 'gm.modules.GeneralConditioner', 'params': {'emb_models': [{'is_trainable': False, 'input_key': 'txt', 'target': 'gm.modules.embedders.modules.FrozenCLIPEmbedder', 'params': {'layer': 'hidden', 'layer_idx': 11, 'version': 'openai/clip-vit-large-patch14'}}, {'is_trainable': False, 'input_key': 'txt', 'target': 'gm.modules.embedders.modules.FrozenOpenCLIPEmbedder2', 'params': {'arch': 'ViT-bigG-14-Text', 'freeze': True, 'layer': 'penultimate', 'always_return_pooled': True, 'legacy': False, 'require_pretrained': False}}, {'is_trainable': False, 'input_key': 'original_size_as_tuple', 'target': 'gm.modules.embedders.modules.ConcatTimestepEmbedderND', 'params': {'outdim': 256}}, {'is_trainable': False, 'input_key': 'crop_coords_top_left', 'target': 'gm.modules.embedders.modules.ConcatTimestepEmbedderND', 'params': {'outdim': 256}}, {'is_trainable': False, 'input_key': 'target_size_as_tuple', 'target': 'gm.modules.embedders.modules.ConcatTimestepEmbedderND', 'params': {'outdim': 256}}]}}, 'first_stage_config': {'target': 'gm.models.autoencoder.AutoencoderKLInferenceWrapper', 'params': {'embed_dim': 4, 'monitor': 'val/rec_loss', 'ddconfig': {'attn_type': 'vanilla', 'double_z': True, 'z_channels': 4, 'resolution': 256, 'in_channels': 3, 'out_ch': 3, 'ch': 128, 'ch_mult': [1, 2, 4, 4], 'num_res_blocks': 2, 'attn_resolutions': [], 'dropout': 0.0}, 'lossconfig': {'target': 'mindspore.nn.Identity'}}}, 'sigma_sampler_config': {'target': 'gm.modules.diffusionmodules.sigma_sampling.DiscreteSampling', 'params': {'num_idx': 1000, 'discretization_config': {'target': 'gm.modules.diffusionmodules.discretizer.LegacyDDPMDiscretization'}}}, 'loss_fn_config': {'target': 'gm.modules.diffusionmodules.loss.StandardDiffusionLoss'}}}, 'optim': {'base_learning_rate': 0.0001, 'optimizer_config': {'target': 'mindspore.nn.AdamWeightDecay', 'params': {'beta1': 0.9, 'beta2': 0.999, 'weight_decay': 0.01, 'eps': 1e-08}}}, 'data': {'per_batch_size': 1, 'total_step': 15000, 'num_parallel_workers': 6, 'python_multiprocessing': False, 'shuffle': True, 'dataset_config': {'target': 'gm.data.dataset.Text2ImageDataset', 'params': {'target_size': 1024, 'transforms': [{'target': 'gm.data.mappers.Resize', 'params': {'size': 1024, 'interpolation': 3}}, {'target': 'gm.data.mappers.Rescaler', 'params': {'isfloat': False}}, {'target': 'gm.data.mappers.AddOriginalImageSizeAsTupleAndCropToSquare'}, {'target': 'gm.data.mappers.Transpose', 'params': {'type': 'hwc2chw'}}]}}}}
Building a Downsample layer with 2 dims.
  --> settings are: 
 in-chn: 320, out-chn: 320, kernel-size: 3, stride: 2, padding: 1
constructing SpatialTransformer of depth 2 w/ 640 channels and 10 heads
WARNING: SpatialTransformer: Found context dims [2048] of depth 1, which does not match the specified 'depth' of 2. Setting context_dim to [2048, 2048] now.
constructing SpatialTransformer of depth 2 w/ 640 channels and 10 heads
WARNING: SpatialTransformer: Found context dims [2048] of depth 1, which does not match the specified 'depth' of 2. Setting context_dim to [2048, 2048] now.
Building a Downsample layer with 2 dims.
  --> settings are: 
 in-chn: 640, out-chn: 640, kernel-size: 3, stride: 2, padding: 1
constructing SpatialTransformer of depth 10 w/ 1280 channels and 20 heads
WARNING: SpatialTransformer: Found context dims [2048] of depth 1, which does not match the specified 'depth' of 10. Setting context_dim to [2048, 2048, 2048, 2048, 2048, 2048, 2048, 2048, 2048, 2048] now.
constructing SpatialTransformer of depth 10 w/ 1280 channels and 20 heads
WARNING: SpatialTransformer: Found context dims [2048] of depth 1, which does not match the specified 'depth' of 10. Setting context_dim to [2048, 2048, 2048, 2048, 2048, 2048, 2048, 2048, 2048, 2048] now.
constructing SpatialTransformer of depth 10 w/ 1280 channels and 20 heads
WARNING: SpatialTransformer: Found context dims [2048] of depth 1, which does not match the specified 'depth' of 10. Setting context_dim to [2048, 2048, 2048, 2048, 2048, 2048, 2048, 2048, 2048, 2048] now.
constructing SpatialTransformer of depth 10 w/ 1280 channels and 20 heads
WARNING: SpatialTransformer: Found context dims [2048] of depth 1, which does not match the specified 'depth' of 10. Setting context_dim to [2048, 2048, 2048, 2048, 2048, 2048, 2048, 2048, 2048, 2048] now.
constructing SpatialTransformer of depth 10 w/ 1280 channels and 20 heads
WARNING: SpatialTransformer: Found context dims [2048] of depth 1, which does not match the specified 'depth' of 10. Setting context_dim to [2048, 2048, 2048, 2048, 2048, 2048, 2048, 2048, 2048, 2048] now.
constructing SpatialTransformer of depth 10 w/ 1280 channels and 20 heads
WARNING: SpatialTransformer: Found context dims [2048] of depth 1, which does not match the specified 'depth' of 10. Setting context_dim to [2048, 2048, 2048, 2048, 2048, 2048, 2048, 2048, 2048, 2048] now.
constructing SpatialTransformer of depth 2 w/ 640 channels and 10 heads
WARNING: SpatialTransformer: Found context dims [2048] of depth 1, which does not match the specified 'depth' of 2. Setting context_dim to [2048, 2048] now.
constructing SpatialTransformer of depth 2 w/ 640 channels and 10 heads
WARNING: SpatialTransformer: Found context dims [2048] of depth 1, which does not match the specified 'depth' of 2. Setting context_dim to [2048, 2048] now.
constructing SpatialTransformer of depth 2 w/ 640 channels and 10 heads
WARNING: SpatialTransformer: Found context dims [2048] of depth 1, which does not match the specified 'depth' of 2. Setting context_dim to [2048, 2048] now.
Unet_lora total params: 2.570366724B, trainable params: 2.90304M.
Turn on recompute, and the unet middle/output blocks will be recomputed.
Initialized embedder #0: FrozenCLIPEmbedder with 123060557 params. Trainable: False
Initialized embedder #1: FrozenOpenCLIPEmbedder2 with 694665770 params. Trainable: False
Initialized embedder #2: ConcatTimestepEmbedderND with 0 params. Trainable: False
Initialized embedder #3: ConcatTimestepEmbedderND with 0 params. Trainable: False
Initialized embedder #4: ConcatTimestepEmbedderND with 0 params. Trainable: False
making attention of type 'vanilla' with 512 in_channels
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
making attention of type 'vanilla' with 512 in_channels
Loading model from ['checkpoints/sd_xl_base_1.0_ms.ckpt']
missing keys:
['conditioner.embedders.0.transformer.text_model.embeddings.position_ids', 'conditioner.embedders.1.model.attn_mask']
Adding mapper Resize as transform #0 to the datapipeline
Adding mapper Rescaler as transform #1 to the datapipeline
Adding mapper AddOriginalImageSizeAsTupleAndCropToSquare as transform #2 to the datapipeline
Adding mapper Transpose as transform #3 to the datapipeline
Dataloader num parallel workers: [6]
scheduler_config not exist, train with base_lr 0.0001
The first step will be compiled for the graph, which may take a long time; You can come back later :).

**Comment by LinF00** [comment]
推理那块有两个问题，一：在线推理加载的模型时间需要优化一下，会切换模型使用，加载模型时间太久了。二：离线模型性能应该会好很多，离线模型大概什么时候能出来

**Comment by zhanghuiyao** [comment]
train.log中未见报错，最后一步还在进行图编译

**Comment by zhanghuiyao** [comment]
> 推理那块有两个问题，一：在线推理加载的模型时间需要优化一下，会切换模型使用，加载模型时间太久了。二：离线模型性能应该会好很多，离线模型大概什么时候能出来

正在规划中，敬请期待。

**Comment by LinF00** [comment]
> train.log中未见报错，最后一步还在进行图编译

看下这个截图报的错
<img width="511" alt="2023-09-12 184828" src="https://github.com/mindspore-lab/mindone/assets/103419218/680e183e-c33c-48eb-bd5d-9a0da4c49fee">

**Comment by zhanghuiyao** [comment]
这个是mindspore版本不匹配导致的，当前代码仅支持 MindSpore 2.0.1，如果只跑推理也可以用MindSpore 2.0版本，后续版本会适配MindSpore 2.1.0；
